src/bin/help.ts

```diff
@@ -0,0 +1,182 @@
+import chalk from "chalk";
+
+import { logPipelessLine } from "../shared/cli/lines.js";
+import { allArgOptions } from "../shared/options/args.js";
+
+interface HelpTextSection {
+	sectionHeading: string;
+	subsections: {
+		flags: Flag[];
+		subheading: string | undefined;
+		warning: string | undefined;
+	}[];
+}
+
+interface Flag {
+	description: string;
+	flag: string;
+	type: string;
+}
+
+interface Option {
+	description: string;
+	docsSection: string;
+	type: string;
+}
+
+function logHelpTextSection(section: HelpTextSection): void {
+	logPipelessLine();
+
+	logPipelessLine(`   ${chalk.black.bgGreenBright(section.sectionHeading)}`);
+
+	for (const subsection of section.subsections) {
+		if (typeof subsection.warning === "string") {
```

Comment:
[Style] Just nipticking here, in cases where `""` isn't ever going to come up I normally default to the more succinct:

```suggestion
		if (subsection.warning) {
```

Was using `typeof` an intentional choice?

Reply:
No, I don't see any reason why the more succinct version won't work!